dstore.go

```diff
@@ -180,19 +180,24 @@ func (d *Datastore) Query(q dsq.Query) (dsq.Results, error) {
 
 	raw = dsq.NaiveOrder(raw, q.Orders...)
 
+	// if we have filters or orders, offset and limit won't have been applied in the query
+	if len(q.Filters) > 0 || len(q.Orders) > 0 {
```

Comment:
I'm not super happy with this arrangement, but it seems worth it to keep queries super fast when there are no filters or orders.